src/router.js

```diff
@@ -25,8 +25,16 @@ import AppSidebar from './views/AppSidebar.vue'
 import Calendar from './views/AppContent/Calendar.vue'
 import Collections from './views/AppContent/Collections.vue'
 
+import { getRootUrl, generateUrl } from '@nextcloud/router'
+
 import { h } from 'vue'
-import { createWebHashHistory, createRouter, RouterView } from 'vue-router'
+import { createWebHistory, createRouter, RouterView } from 'vue-router'
+
+const webRootWithIndexPHP = getRootUrl() + '/index.php'
+const doesURLContainIndexPHP = window.location.pathname.startsWith(webRootWithIndexPHP)
+const base = generateUrl('apps/tasks', {}, {
+	noRewrite: doesURLContainIndexPHP,
+})
```

Comment:
@susnux @Antreesy I saw that other apps (e.g. Calendar, see https://github.com/nextcloud/calendar/blob/b93fed45c3c61faa82c7fe1f3aa27c4d3f3ee792/src/router.js#L39-L43) do it like this, but it seems a bit complicated to me. Is there an easier way?

Reply:
Haven't seen anything like it. 
But you could check the latest release of nextcloud/router, getRootUrl was rewritten, perhaps, it covers that

Reply:
Thanks for the hint. I had a look, but it doesn't give exactly the same result. I guess it works as it is now. We can always improve later on.